cfme/infrastructure/host.py

```diff
@@ -558,3 +564,35 @@ def find_quadicon(host, do_not_navigate=False):
             return quadicon
     else:
         raise HostNotFound("Host '{}' not found in UI!".format(host))
+
+
+def register(provider, file_path, cancel=False):
+    """Register new nodes (Openstack)
+
+    Input file path of json file with new node details, navigation has to
+    be from a specific provider
+    """
+    provider.load_details()
+    sel.click(InfoBlock.element("Relationships", "Nodes"))
+    tb.select('Configuration', 'Register Nodes')
+    my_form = {'file': file_path}
+    fill(register_nodes_form, my_form, action=register_nodes_form.register)
+
+
+def host_exist(provider, name='my_node'):
+    """" registered imported host exist
+    provider - provider instance
+    name - by default name is my_name
+    Input provider, name of the new node, looking for the host in Ironic
+    clients, compare the record found with hosts list in CFME DB
+    """
+    nodes = provider.mgmt.list_node()
+    nodes_dict = {i.name: i for i in nodes}
+    query = provider.appliance.db.session.query(
+        provider.appliance.db['hosts'], 'guid')
+    node_uuid = str(nodes_dict[name])
+    for db_node in query.all():
```

Comment:
Why do you iterate through whole query result if you return after boolean value first element. It might be `query.first()` will be enough.